### PR TITLE
Fix preact state delay

### PR DESCRIPTION
**Pull request by pentacular**
Looks like preact doesn't propagate state changes immediately, so we need to advise the function directly in some cases.

Also seem to have fixed the widget size computation problem.

### CODE REVIEW

File: api/shape/view.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ export const baseView =
     wireframe = false,
     prepareView = (x) => x,
     inline,
-    width = 1024,
```
Comment (BarbourSmith) [inline api/shape/view.js:21]: Does this make views smaller by default?

File: es6/jsxcad-ui-notebook.js [hunk 2]
```diff
@@ -2767,10 +2767,12 @@ const toDomElement = (notebook = [], { onClickView } = {}) => {
       Object.assign(entry, note.define.data);
     }
     if (note.view) {
-      const { url, openView } = note;
+      const { url, openView, view } = note;
+      const { height, width } = view;
       const image = document.createElement('img');
       image.style.display = 'block';
-      image.style.height = `${21 * 13}px`;
+      image.style.height = `${height}px`;
```
Comment (BarbourSmith) [inline es6/jsxcad-ui-notebook.js:2774]: Nice!

File: ui/v1/ui.js [hunk 3]
```diff
@@ -304,6 +305,8 @@ class Ui extends PureComponent {
             def.domElement.appendChild(document.createTextNode(entry.hash));
             def.domElement.appendChild(element);
             console.log(`Marking ${entry.hash} in ${id}`);
+            // Let the node attach.
+            await sleep(0);
```
Comment (BarbourSmith) [inline ui/v1/ui.js:309]: How does this work? Is the point that sleep is non-blocking here so that thread is able to handle other calls on the stack?

**Review by BarbourSmith — APPROVED** [review]
Looks good to me 🙂

File: api/shape/view.js (hunk 1 above)
Comment (pentacular) [reply api/shape/view.js:21]: Yes, but I think they were rendering smaller anyhow -- when I fixed the sizes they got very large. :)

File: ui/v1/ui.js (hunk 3 above)
Comment (pentacular) [reply ui/v1/ui.js:309]: Ah, I thought I'd removed that.

The idea was to let the browser render the dom nodes -- the active code is in

      await new Promise((resolve, reject) => {
        requestAnimationFrame(resolve);
      });

in JsEditor

Comment (pentacular) [reply ui/v1/ui.js:309]: Without this we get a zero offsetHeight on the widget element.

**Pull request merged**